client/components/data/query-jetpack-connection/index.jsx

```diff
@@ -12,13 +12,15 @@ import { requestJetpackConnectionStatus } from 'state/jetpack/connection/actions
 
 class QueryJetpackConnection extends Component {
 	static propTypes = {
-		siteId: PropTypes.number.isRequired,
+		siteId: PropTypes.number,
 		requestingJetpackConnectionStatus: PropTypes.bool,
 		requestJetpackConnectionStatus: PropTypes.func
 	};
 
 	componentWillMount() {
-		this.request( this.props );
+		if ( this.props.siteId !== null ) {
```

Comment:
Let's do that check in `this.request` instead - see https://github.com/Automattic/wp-calypso/blob/master/client/components/data/query-jetpack-user-connection/index.jsx#L33 for example.